operator/controllers/common/test_utils.go

```diff
@@ -10,32 +10,42 @@ import (
 	"go.opentelemetry.io/otel/metric/unit"
 	"go.opentelemetry.io/otel/sdk/metric"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 func AddApp(c client.Client, name string) error {
 	app := &lfcv1alpha2.KeptnApp{
 		TypeMeta: metav1.TypeMeta{},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      name,
-			Namespace: "default",
+			Name:       name,
+			Namespace:  "default",
+			Generation: 1,
 		},
 		Spec: lfcv1alpha2.KeptnAppSpec{
 			Version: "1.0.0",
 		},
 		Status: lfcv1alpha2.KeptnAppStatus{},
 	}
 	return c.Create(context.TODO(), app)
+}
 
+func UpdateAppRevision(c client.Client, name string, revision uint) error {
+	app := &lfcv1alpha2.KeptnApp{}
+	c.Get(context.TODO(), types.NamespacedName{Namespace: "default", Name: name}, app)
+	app.Spec.Revision = revision
+	app.Generation = int64(revision)
```

Comment:
```suggestion
	app.Generation = revision
```

Reply:
generation is int64, it needs to be there